### PR TITLE
Fix issue with inline-formatting changing indentation of multi-line constructs

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/16198

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide .

The problem before was that we were copying trivia from one construct to another *including* indentation.  This was causing a node's first line to be double-indented.  When teh formatted then moved it back into place, it also adjusted all following parts of hte statement, making them appear like they'd been shifted left unnecessarily.  Now we copy over the trivia *not* counting hte indentation.

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs [hunk 1]
```diff
@@ -92,16 +92,14 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 var block = (BlockSyntax)localDeclarationStatement.Parent;
                 var declarationIndex = block.Statements.IndexOf(localDeclarationStatement);
 
-                // Comments on the local declaration will move to the next statement.  Format it
-                // so that it the comments show up properly.
+                // Trivia on the local declaration will move to the next statement.
+                // use the callback form as the next statement may be hte place where we're
```
Comment (Pilchie) [inline src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs]: Capitalize Use, and fix spelling of hte.

File: src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs [hunk 2]
```diff
@@ -165,6 +163,30 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             editor.ReplaceNode(identifier, declarationExpression);
         }
 
+        private static SyntaxNode MoveNonIndentationTrivia(
+            SyntaxNode from, SyntaxNode to)
+        {
+            // get all the preceding trivia from the 'from' node, not counting the leading
+            // indentation trivia is has.
```
Comment (Pilchie) [inline src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs]: "it" has.

File: src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs [hunk 3]
```diff
@@ -165,6 +163,30 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             editor.ReplaceNode(identifier, declarationExpression);
         }
 
+        private static SyntaxNode MoveNonIndentationTrivia(
+            SyntaxNode from, SyntaxNode to)
+        {
+            // get all the preceding trivia from the 'from' node, not counting the leading
```
Comment (Pilchie) [inline src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs]: Capitalize.

File: src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs [hunk 4]
```diff
@@ -165,6 +163,30 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
...
+            // indentation trivia is has.
+            var finalTrivia = from.GetLeadingTrivia().ToList();
+            while (finalTrivia.Count > 0 && finalTrivia.Last().Kind() == SyntaxKind.WhitespaceTrivia)
+            {
+                finalTrivia.RemoveAt(finalTrivia.Count - 1);
+            }
+
+            // Also, add on hte trailing trivia if there are trailing comments.
```
Comment (Pilchie) [inline src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs]: `hte`

Comment (Pilchie) [inline src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs]: `hte`

File: src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs (hunk 1 above)
Comment (Pilchie) [inline src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs]: Also, it looks like it's not it "may" be the place - it's always the place isn't it?

Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs]: No.  Inline decl can inline to many statements in teh future.  For example:

```c#
int i;
int j;
Foo(out i);
```

**Comment by CyrusNajmabadi** [comment]
Kevin, how about you just write the comments for my code? :)   Wouldn't it be easier to just edit things in github?

**Comment by Pilchie** [comment]
https://marketplace.visualstudio.com/items?itemName=EWoodruff.VisualStudioSpellCheckerVS2017andLater

**Comment by CyrusNajmabadi** [comment]
Speeling has been ficksed

**Pull request merged**